### PR TITLE
Add a Bazel build for cartographer

### PR DESCRIPTION
**Pull request by drigz**
This does not include cartographer_grpc, nor any of the binaries
(*_main.cc). It has been tested with Bazel 0.9.0 on Ubuntu 14.04 with
the latest (at time of writing) dazel/dazel container, by running:

```
bazel build //...
bazel test //...
```

While this is a large CL, the files under third_party are not new - they have been through internal review, and this is their open-source release. The other files come to 212 lines of code.

### CODE REVIEW

File: tools/bazel.rc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2018 The Cartographer Authors
```
Comment (cschuet) [inline tools/bazel.rc]: Is it important that this is in tools? We have a cmake subdirectory. Should we rather have this file in a bazel subdirectory?

Comment (cschuet) [reply tools/bazel.rc]: As discussed this one needs to stay but others have gone into the subfolder.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

File: tools/bazel.rc (hunk 1 above)
Comment (drigz) [reply tools/bazel.rc]: https://github.com/bazelbuild/bazel/issues/4502

**Pull request merged**